[pkg] add generic ZipFile Reader/Writer

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #74929
* __->__ #72237

add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

PaliC committed:
[pkg] add generic ZipFile Reader/Writer

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"




Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer + add change persistent_id to __reduce_package__ for storage"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

This PR also adds changes in order to allow for torchscript model serialization without importing torch into torch.package.PackageImporter or torch.package.PackageExporter.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

suo commented:
can we split out the "add generic ZipFile Reader/Writer" and the "add change persistent_id to __reduce_package__ for storage" changes? They seem unrelated and it would make things easier to review.

suo reviewed (changes requested):
back to you for splitting

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer + add change persistent_id to __reduce_package__ for storage"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

This PR also adds changes in order to allow for torchscript model serialization without importing torch into torch.package.PackageImporter or torch.package.PackageExporter.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

suo reviewed (changes requested):
Not sure this PR achieves the outcome it intends to—we want to create an abstract ZipReader/Writer interface various clients can use, so that we can swap out the implementation for one backed by Python's zipfile module. But here, our clients are always expecting a TorchScriptZipFileWriter—if you pass it a different one, things will break. 

In order to accomplish the goal of abstraction, all clients of the abstract ZipReader/Writer interface must *only* make use of functionality exposed by the interface.

PaliC commented on torch/package/_zip_file.py:
```diff
@@ -0,0 +1,91 @@
+    def has_record(self, path):
+        return path in records
+
+    def write_record(self, file_name, str_or_bytes, size=None):
```
nope, this was a typo

PaliC commented on torch/package/package_exporter.py:
```diff
@@ -35,6 +35,10 @@
 from .importer import Importer, OrderedImporter, sys_importer
+from ._zip_file_torchscript import TorchScriptPackageZipFileWriter
+from ._zip_file import PackageZipFileWriter
+import inspect
```
do you mean inspect? If so it actually gets used to check if an objects a class. Though for this pr probably worth removing.

PaliC commented on torch/package/package_exporter.py:
```diff
@@ -181,6 +185,7 @@ def __init__(
         self,
         f: Union[str, Path, BinaryIO],
         importer: Union[Importer, Sequence[Importer]] = sys_importer,
+        zip_file_reader_type: Type[PackageZipFileWriter] = TorchScriptPackageZipFileWriter
```
So this was more for PR splitting.  https://github.com/pytorch/pytorch/pull/72246 and https://github.com/pytorch/pytorch/pull/72247 replace TorchScriptPackageZipFileWriter with DefaultPackageZipFileWriter and create the necessary shim for torch. Is there a better way to split things up?

PaliC commented on torch/package/_zip_file_torchscript.py:
```diff
@@ -0,0 +1,52 @@
+import torch
+from ._zip_file import PackageZipFileReader, PackageZipFileWriter
+
+class TorchScriptPackageZipFileWriter(PackageZipFileWriter):
```
yeah this is true. Refer to this comment https://github.com/pytorch/pytorch/pull/72237#discussion_r804266144. I might be splitting this diff wrong. The idea was just to create the baseline abstractions and wrappers needed for the zip file. The integration happens later on.

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented on torch/package/_zip_file.py:
```diff
@@ -0,0 +1,91 @@
+        return super().read(name)
+
+    def has_record(self, path):
+        return path in records
```
I'll add another PR to add the unit tests, worth finalizing this stuff first lol

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

d4l3k commented on test/package/package_a/test_module.py:
```diff
@@ -37,6 +37,7 @@ def forward(self, x):
 class ModWithTwoSubmodsAndTensor(torch.nn.Module):
     def __init__(self, tensor, sub_mod_0, sub_mod_1):
         super().__init__()
+        self.name = 'ModWithTwoSubmodsAndTensor'
```
what's this for?

d4l3k commented on torch/package/package_exporter.py:
```diff
@@ -202,8 +206,11 @@ def __init__(
-        self.zip_file = torch._C.PyTorchFileWriter(f)
-        self.zip_file.set_min_version(6)
+        self.zip_file = zip_file_reader_type(f)
+        assert isinstance(self.zip_file, TorchScriptPackageZipFileWriter)
```
asserting here seems to make zip_file_reader_type useless since you can only have a single type, is this intentional?

PaliC replied:
This is one of those make mypy happy type of assertions :( Is there a better way to do this?

d4l3k commented on torch/jit/_script.py:
```diff
@@ -342,15 +343,19 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
-    if not isinstance(importer.zip_reader, torch._C.PyTorchFileReader):
+
+    # if 'storage_context' not in importer.external_registry:
+    #     importer.external_registry['storage_context'] = torch._C.DeserializationStorageContext()
```
remove

d4l3k commented on torch/jit/_script.py:
```diff
@@ -342,15 +343,19 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
     cpp_module = torch._C._import_ir_module_from_package(
         cu,
-        importer.zip_reader,
+        importer.zip_reader.zip_file_reader,
```
does this _need_ the C++ implementation? wondering if using the abstract classes here is just adding more layers without providing much simplification

PaliC replied:
Yeah unfortunately, I think it does. At least I haven't figured out a nice way to avoid invoking it.

d4l3k commented on torch/package/package_exporter.py:
```diff
@@ -202,8 +206,11 @@ def __init__(
+        self.zip_file = zip_file_reader_type(f)
+        assert isinstance(self.zip_file, TorchScriptPackageZipFileWriter)
+        self.script_module_serializer = torch._C.ScriptModuleSerializer(self.zip_file.zip_file_writer)
+        self.storage_context = self.script_module_serializer.storage_context()
```
seems like these torchscript specifics should be moved into the TorchScriptPackageZipFileWriter so it can handle this when torch is present only

d4l3k commented on torch/package/package_importer.py:
```diff
@@ -72,13 +73,12 @@ def __init__(
             self.filename = "<binary>"
-            self.zip_reader = torch._C.PyTorchFileReader(file_or_buffer)
-
+            self.zip_reader = TorchScriptPackageZipFileReader(file_or_buffer)
```
instead of hardcoding this can we have at the top of the file:

```
HAS_TORCH = False
try:
   import torch
   HAS_TORCH = True
except:
  pass
  
if HAS_TORCH:
  PackageZipFileReader = TorchScriptPackageZipFileReader
else:
  PackageZipFileReader = PythonPackageZipFileReader
```

And then we can just use PackageZipFileReader everywhere and simplify this logic up a lot

PaliC replied:
Instead of this I ended up just doing subclassing for the PackageImporter/Exporter to avoid the fragility of the flags. Is there a reason this approach would be preferable?

d4l3k commented on torch/package/package_importer.py:
```diff
@@ -193,7 +193,7 @@ def load_tensor(dtype, size, key, location, restore_location):
                     ".data/" + name, size, dtype
                 )
-                if isinstance(self.zip_reader, torch._C.PyTorchFileReader):
+                if isinstance(self.zip_reader, TorchScriptPackageZipFileReader):
```
seems like something that could be moved into TorchScriptPackageZipFileReader

might want to iterate on that interface if we can't support this

d4l3k replied:
this one is still here but it's a lot cleaner

d4l3k commented on torch/jit/_script.py:
```diff
@@ -342,15 +343,19 @@ def unpackage_script_module(importer: PackageImporter, script_module_id: str) ->
+    # if 'storage_context' not in importer.external_registry:
+    #     importer.external_registry['storage_context'] = torch._C.DeserializationStorageContext()
+
+    if not isinstance(importer.zip_reader, TorchScriptPackageZipFileReader):
```
can we make it so TorchScriptPackageZipFileReader calls this method instead of the other way around? That would make it easier for us to split package out from core and cleans up the dependency hierarchy

d4l3k reviewed (commented):
this is getting better but there's still some leaky abstractions that I think we can clean up

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

d4l3k reviewed (approved):
Thanks for cleaning that up! It would be nice to add python static types to the newly added files. mypy runs in pytorch repo right?

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
add unit tests for the new zipfile classes on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

d4l3k reviewed (approved):
reapproving

PaliC committed:
Update on "[pkg] add generic ZipFile Reader/Writer"


add a generic zip file reader/writer to torch.package in order to get rid of dependency on torch for non torchscript / tensor related usages of package. This also enables users to create a derived class from the zip file reader/writer classes to have their own serialization/deserialization if it's desired for performance needs.

Differential Revision: [D33970688](https://our.internmc.facebook.com/intern/diff/D33970688)

[ghstack-poisoned]

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33970688).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423079).

PaliC commented:
@PaliC has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35423079).

mruberry commented:
Unlanding this stack. Unfortunately it looks like this broke trunk / macos-11-py3-x86-64 / test (default, 1, 2, macos-11). See an example log here: https://github.com/pytorch/pytorch/runs/5863579163?check_suite_focus=true.
